File: data/derived_cot/rq1_traces/translation/r1/java_to_cpp/code_output/DataStatistics.cpp
#include <vector>
#include <algorithm>
#include <map>
#include <cmath>
#include <stdexcept>

class DataStatistics {
public:
    double mean(const std::vector<int>& data) {
        if (data.empty()) {
            return 0.0;
        }
        double sum = 0.0;
        for (int num : data) {
            sum += num;
        }
        double avg = sum / static_cast<double>(data.size());
        double temp = avg * 100.0;
        temp = std::round(temp);
        return temp / 100.0;
    }

    double median(std::vector<int> data) {
        if (data.empty()) {
            throw std::out_of_range("Array index out of bounds");
        }
        std::sort(data.begin(), data.end());
        size_t n = data.size();
        if (n % 2 == 0) {
            size_t mid = n / 2;
            double midValue = (static_cast<double>(data[mid-1]) + data[mid]) / 2.0;
            double temp = midValue * 100.0;
            temp = std::round(temp);
            return temp / 100.0;
        } else {
            return static_cast<double>(data[n/2]);
        }
    }

    std::vector<int> mode(const std::vector<int>& data) {
        if (data.empty()) {
            throw std::runtime_error("No such element");
        }
        std::map<int, long> frequencyMap;
        for (int num : data) {
            frequencyMap[num]++;
        }

        long maxCount = 0;
        for (const auto& pair : frequencyMap) {
            if (pair.second > maxCount) {
                maxCount = pair.second;
            }
        }

        std::vector<int> modes;
        for (const auto& pair : frequencyMap) {
            if (pair.second == maxCount) {
                modes.push_back(pair.first);
            }
        }

        std::sort(modes.begin(), modes.end());
        return modes;
    }
};

File: data/derived_cot/rq3_prompting/results/translation/qwen/cpp_to_py/code_output/DataStatistics.py
class DataStatistics:
    def mean(self, data):
        if not data:
            raise ValueError("Data vector is empty.")
        total = sum(data)
        return round(total / len(data) * 100) / 100

    def median(self, data):
        if not data:
            raise ValueError("Data vector is empty.")
        data_sorted = sorted(data)
        n = len(data_sorted)
        if n % 2 == 0:
            middle_sum = (data_sorted[n//2 - 1] + data_sorted[n//2]) / 2
            return round(middle_sum * 100) / 100
        else:
            return data_sorted[n//2]

    def mode(self, data):
        if not data:
            raise ValueError("Data vector is empty.")
        count_map = {}
        for num in data:
            count_map[num] = count_map.get(num, 0) + 1
        max_count = max(count_map.values())
        modes = [num for num, count in count_map.items() if count == max_count]
        return modes